[ASM] - EXPANDR -  1578

johnnywilkes opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-hq.paloaltonetworks.local/browse/EXPANDR-1578

## Description
Remediation Path Rules is a new feature in the Xpanse product with UI elements that allows customers’ to express their preferences about how they address different types of risks in different situations.  Content involves a new integration command to pull these rules and an automation to determine is a match was found for the current alert.  I have included a short demo in the registration form but will also provide a live one if necessary.

## Screenshots
![Cortex_ASM_-_Remediation_Path_Rules](https://user-images.githubusercontent.com/32227961/217071710-41f15377-0c9d-4a9e-87cf-c00fbfd7517e.png)

![Cortex_ASM_-_Decision](https://user-images.githubusercontent.com/32227961/217071664-b9b49e4b-87b4-4045-81f9-a7c81c4d04a2.png)

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=24380) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=24380) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: johnnywilkes<br/>:x: songcelia88<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=24380) it.</sub>

johnnywilkes commented:
@ilaner will share video with you

BigEasyJ commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation_test.py:
```diff
@@ -0,0 +1,249 @@
+
+    alert_context = {"severity": None}
+    cond = {"field": "severity", "operator": "eq", "value": "MEDIUM"}
+    assert evaluate_criteria(cond, alert_context) is False
```
```suggestion
@pytest.mark.parametrize(
    "test_condition, test_alert_context, expected_result",
    [
      ({"field": "severity", "operator": "eq", "value": "LOW"}, ALERT_CONTEXT, True),
      ({"field": "severity", "operator": "eq", "value": "HIGH"},ALERT_CONTEXT, True),  
      ({"field": "severity", "operator": "eq", "value": "MEDIUM"}, {"severity": None}, False),
    ],
)
def test_evaluate_criteria_severity(test_condition, test_alert_context, expected_result) :
    assert evaluate_criteria(test_condition, test_alert_context) is expected_result
```

BigEasyJ replied:
A similar approach could be taken with the following tests as well.

BigEasyJ commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.py:
```diff
@@ -0,0 +1,225 @@
+
+    if operator == "eq":
+        # severity
+        if field == "severity":
```
It might help us to reduce complexity here in case it gets bigger in the future. Is it possible to make this nested `if field == "severity"` statement not nested by moving down. That way we could use something like:

```suggestion
    if operator != "eq":
        demisto.info(f"Condition Operator {operator} is not supported at this time.")
        return False
```
and then it can exit early.

johnnywilkes replied:
We can change this later if needed but for now all operators are "eq" so we shouldn't see it match on anything else for now.

BigEasyJ commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.py:
```diff
@@ -0,0 +1,225 @@
+            match = False
+        else:
+            # check criteria conjunction
+            if rule.get("criteria_conjunction") == "AND":
```
Can this be changed to an `elif`?

johnnywilkes replied:
possibly, but I don't really see that making the code more simple/readable and not sure what the ending `else` would then be as well

BigEasyJ commented on Packs/CortexAttackSurfaceManagement/Integrations/CortexAttackSurfaceManagement/CortexAttackSurfaceManagement_test.py:
```diff
@@ -255,3 +249,41 @@ def test_get_asset_internet_exposure_command(requests_mock):
+
+    response = list_remediation_rule_command(client, args)
+
+    assert response.outputs == REMEDIATION_RULES_RESULTS
```
Not necessary to do for this, but if you ever need to increase code coverage you can also import `CommandResults` from CommonServerPython. You can create a CommandResults object in the test and set it to a variable and check you function output to it. 

Then you could assert it like this:
```python
assert command_results_from_your_function.to_context() == expected_command_results.to_context()
```

johnnywilkes replied:
thanks for the input, i can keep this in mind for later

johnnywilkes commented on Packs/CortexAttackSurfaceManagement/Integrations/CortexAttackSurfaceManagement/CortexAttackSurfaceManagement.py:
```diff
@@ -14,11 +14,25 @@ class Client(BaseClient):
     """
 
-    def __init__(self, base_url, verify, proxy, headers, auth):
+    def __init__(self, base_url, verify, proxy, headers):
```
it was removed as part of feedback from https://github.com/demisto/content/pull/24132 (same integration but added commands).  It isn't being reference, so no need to have it.

johnnywilkes commented on Packs/CortexAttackSurfaceManagement/Integrations/CortexAttackSurfaceManagement/CortexAttackSurfaceManagement.yml:
```diff
@@ -327,7 +327,50 @@ script:
       type: Unknown
-  dockerimage: demisto/python3:3.10.8.37753
+  - arguments:
+    - description: A string representing the ASM Rule ID you want to get association remediation path rules for.
```
thank you for finding this

johnnywilkes commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.py:
```diff
@@ -0,0 +1,225 @@
+import demistomock as demisto  # noqa: F401
```
just verified that this is added through demisto-sdk and will not show up in UI

johnnywilkes commented on Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport.py:
```diff
@@ -1,11 +1,10 @@
 from base64 import b64decode
+from typing import Any, Dict, List
+
+import demistomock as demisto  # noqa: F401
```
just verified that this is added through demisto-sdk and will not show up in UI

BigEasyJ reviewed (approved):
Approved

johnnywilkes commented:
Thank you @BigEasyJ for approval
@ilaner , can you please assign @ShirleyDenkberg for documentation review?

BigEasyJ commented:
Changed to commit 40cb403 for testing purposes only. Changed back to the original state of 54c780f.

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Integrations/CortexAttackSurfaceManagement/CortexAttackSurfaceManagement.yml:
```diff
@@ -327,14 +327,54 @@ script:
       type: Unknown
-  dockerimage: demisto/python3:3.10.8.37753
+  - arguments:
+    - description: A string representing the ASM Rule ID you want to get the associated remediation path rules for.
```
```suggestion
    - description: A string representing the ASM rule ID you want to get the associated remediation path rules for.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Integrations/CortexAttackSurfaceManagement/CortexAttackSurfaceManagement.yml:
```diff
@@ -327,14 +327,54 @@ script:
+      description: Remediation path rule description.
+      type: String
+    - contextPath: ASM.RemediationRule.attack_surface_rule_id
+      description: Association ASM rule ID for Remediation path rules.
```
```suggestion
      description: Association ASM rule ID for the remediation path rules.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Integrations/CortexAttackSurfaceManagement/CortexAttackSurfaceManagement.yml:
```diff
@@ -327,14 +327,54 @@ script:
+      description: Array of remediation path rule criteria.
+      type: Unknown
+    - contextPath: ASM.RemediationRule.criteria_conjunction
+      description: Whether criteria is processes with AND or OR.
```
```suggestion
      description: Whether criteria is processed with AND or OR.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Integrations/CortexAttackSurfaceManagement/README.md:
```diff
@@ -1049,7 +1049,78 @@ Get internet exposure asset details according to the asset ID.
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| asm_rule_id | A string representing the ASM Rule ID you want to get association remediation path rules for. | Required | 
```
```suggestion
| asm_rule_id | A string representing the ASM rule ID you want to get the association remediation path rules for. | Required | 
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Integrations/CortexAttackSurfaceManagement/README.md:
```diff
@@ -1049,7 +1049,78 @@ Get internet exposure asset details according to the asset ID.
+| ASM.RemediationRule.description | String | Remediation path rule description. | 
+| ASM.RemediationRule.attack_surface_rule_id | String | Association ASM rule ID for Remediation path rules. | 
+| ASM.RemediationRule.criteria | Unknown | Array of remediation path rule criteria. | 
+| ASM.RemediationRule.criteria_conjunction | String | Whether criteria is processes with AND or OR. | 
```
```suggestion
| ASM.RemediationRule.criteria_conjunction | String | Whether criteria is processed with AND or OR. | 
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert.yml:
```diff
@@ -2966,57 +2670,267 @@ tasks:
+      id: 2bdb4a9e-2e38-4560-80de-52908d6873e6
+      version: -1
+      name: Cortex ASM - Remediation Path Rules
+      description: This playbook returns "RemediationAction" options based return from Remediation Path Rules API, or defaults to data collection task options from "Cortex ADM - Decision" subplaybook.
```
```suggestion
      description: This playbook returns "RemediationAction" options based on the return from the Remediation Path Rules API, or defaults to data collection task options from "Cortex ADM - Decision" sub-playbook.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ASM_Alert_README.md:
```diff
@@ -4,32 +4,32 @@ This playbook handles ASM alerts by enriching asset information and providing a
-| AutomatedRemediation | Decide whether to potentially skip analyst intervention and conduct automated remediation if meeting the criteria specified here: https://cortex.marketplace.pan.dev/marketplace/details/CortexAttackSurfaceManagement/ <br/><br/>Default is "False" \(don't do automated remediation\).  Set "True" to turn this feature on. | False | Required |
 | OwnerNotificationSubject | Subject of the notification \(email or ticket\) sent to potential service owner. | A new security risk was identified on an external service owned by your team | Required |
-| OwnerNotificationBody | Body of the notification \(email or ticket\) sent to potential service owner. | Infosec identified a security risk on an external service potentially owned by your team: ${alert.name}&lt;br&gt;&lt;br&gt;<br/>Description: ${alert.details}<br/>&lt;br&gt;&lt;br&gt;<br/><br/> | Required |
+| OwnerNotificationBody | Body of the notification \(email or ticket\) sent to potential service owner. | Infosec identified a security risk on an external service potentially owned by your team: ${alert.name}<br/><br/>Description: ${alert.details}<br/><br/> | Required |
```
```suggestion
| OwnerNotificationBody | Body of the notification \(email or ticket\) sent to a potential service owner. | Infosec identified a security risk on an external service potentially owned by your team: ${alert.name}<br/><br/>Description: ${alert.details}<br/><br/> | Required |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision.yml:
```diff
@@ -0,0 +1,345 @@
+version: -1
+name: Cortex ASM - Decision
+description: |-
+  This playbook returns "RemediationAction" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is setup.  Possible return values are:
```
```suggestion
  This playbook returns "RemediationAction" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is set up.  Possible return values are:
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision.yml:
```diff
@@ -0,0 +1,345 @@
+name: Cortex ASM - Decision
+description: |-
+  This playbook returns "RemediationAction" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is setup.  Possible return values are:
+  - Prompt base - data collection task with only email/manual options
```
```suggestion
  - Prompt base - data collection task with only email/manual options.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision.yml:
```diff
@@ -0,0 +1,345 @@
+description: |-
+  This playbook returns "RemediationAction" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is setup.  Possible return values are:
+  - Prompt base - data collection task with only email/manual options
+  - Prompt all options - data collection task with all options (meets Automated Remediation requirements and ServiceNow enabled)
```
```suggestion
  - Prompt all options - data collection task with all options (meets Automated Remediation requirements and ServiceNow is enabled).
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision.yml:
```diff
@@ -0,0 +1,345 @@
+  - Prompt base - data collection task with only email/manual options
+  - Prompt all options - data collection task with all options (meets Automated Remediation requirements and ServiceNow enabled)
+  - Prompt no snow - all options except ServiceNow (Automated Remediation requirements met)
+  - Prompt no ar - all options except Automated Remediation (ServiceNow enabled)
```
```suggestion
  - Prompt no ar - all options except Automated Remediation (ServiceNow is enabled).
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision.yml:
```diff
@@ -0,0 +1,345 @@
+  This playbook returns "RemediationAction" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is setup.  Possible return values are:
+  - Prompt base - data collection task with only email/manual options
+  - Prompt all options - data collection task with all options (meets Automated Remediation requirements and ServiceNow enabled)
+  - Prompt no snow - all options except ServiceNow (Automated Remediation requirements met)
```
```suggestion
  - Prompt no snow - all options except ServiceNow (Automated Remediation requirements met).
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision.yml:
```diff
@@ -0,0 +1,345 @@
+      id: a6491b8e-4047-4d8d-86f0-766269ffc3bc
+      version: -1
+      name: Which option to return?
+      description: Checks playbooks inputs to determine if "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) were met and if ServiceNowV2 integration is setup.
```
```suggestion
      description: Checks playbook inputs to determine if "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) were met and if ServiceNowV2 integration is set up.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision.yml:
```diff
@@ -0,0 +1,345 @@
+- key: AutoRemediationRequirements
+  value: {}
+  required: true
+  description: Meetings automated remediation requirements (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements).
```
```suggestion
  description: Meets automated remediation requirements (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements).
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision.yml:
```diff
@@ -0,0 +1,345 @@
+  playbookInputQuery:
+outputs:
+- contextPath: RemediationAction
+  description: Follow up action to perform (variety of data collection tasks)
```
```suggestion
  description: Follow up action to perform (variety of data collection tasks).
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision_README.md:
```diff
@@ -0,0 +1,39 @@
+This playbook returns "RemediationAction" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is setup.  Possible return values are:
```
```suggestion
This playbook returns "RemediationAction" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is set up.  Possible return values are:
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision_README.md:
```diff
@@ -0,0 +1,39 @@
+This playbook returns "RemediationAction" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is setup.  Possible return values are:
+- Prompt base - data collection task with only email/manual options
```
```suggestion
- Prompt base - data collection task with only email/manual options.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision_README.md:
```diff
@@ -0,0 +1,39 @@
+This playbook returns "RemediationAction" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is setup.  Possible return values are:
+- Prompt base - data collection task with only email/manual options
+- Prompt all options - data collection task with all options (meets Automated Remediation requirements and ServiceNow enabled)
```
```suggestion
- Prompt all options - data collection task with all options (meets Automated Remediation requirements and ServiceNow is enabled).
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision_README.md:
```diff
@@ -0,0 +1,39 @@
+This playbook returns "RemediationAction" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is setup.  Possible return values are:
+- Prompt base - data collection task with only email/manual options
+- Prompt all options - data collection task with all options (meets Automated Remediation requirements and ServiceNow enabled)
+- Prompt no snow - all options except ServiceNow (Automated Remediation requirements met)
```
```suggestion
- Prompt no snow - all options except ServiceNow (Automated Remediation requirements are met).
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision_README.md:
```diff
@@ -0,0 +1,39 @@
+- Prompt base - data collection task with only email/manual options
+- Prompt all options - data collection task with all options (meets Automated Remediation requirements and ServiceNow enabled)
+- Prompt no snow - all options except ServiceNow (Automated Remediation requirements met)
+- Prompt no ar - all options except Automated Remediation (ServiceNow enabled)
```
```suggestion
- Prompt no ar - all options except Automated Remediation (ServiceNow is enabled).
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision_README.md:
```diff
@@ -0,0 +1,39 @@
+This playbook does not use any integrations.
+
+### Scripts
+* Set
```
```suggestion
Set
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision_README.md:
```diff
@@ -0,0 +1,39 @@
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| ServiceNowv2Enabled | ServiceNow v2 Integration enabled. |  | Required |
```
```suggestion
| ServiceNowv2Enabled | ServiceNow v2 integration enabled. |  | Required |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision_README.md:
```diff
@@ -0,0 +1,39 @@
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| ServiceNowv2Enabled | ServiceNow v2 Integration enabled. |  | Required |
+| AutoRemediationRequirements | Meetings automated remediation requirements \(https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement\#automated-remediation-requirements\). |  | Required |
```
```suggestion
| AutoRemediationRequirements | Meets automated remediation requirements \(https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement\#automated-remediation-requirements\). |  | Required |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision.yml:
```diff
@@ -0,0 +1,345 @@
+- key: ServiceNowv2Enabled
+  value: {}
+  required: true
+  description: ServiceNow v2 Integration enabled.
```
```suggestion
  description: ServiceNow v2 integration enabled.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Decision_README.md:
```diff
@@ -0,0 +1,39 @@
+
+| **Path** | **Description** | **Type** |
+| --- | --- | --- |
+| RemediationAction | Follow up action to perform \(variety of data collection tasks\) | string |
```
```suggestion
| RemediationAction | Follow up action to perform \(variety of data collection tasks\). | string |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Remediation_Path_Rules.yml:
```diff
@@ -0,0 +1,1010 @@
+id: Cortex ASM - Remediation Path Rules
+version: -1
+name: Cortex ASM - Remediation Path Rules
+description: This playbook returns "RemediationAction" options based return from Remediation Path Rules API, or defaults to data collection task options from "Cortex ADM - Decision" subplaybook.
```
```suggestion
description: This playbook returns "RemediationAction" options based on the return from the Remediation Path Rules API, or defaults to data collection task options from the "Cortex ADM - Decision" sub-playbook.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Remediation_Path_Rules.yml:
```diff
@@ -0,0 +1,1010 @@
+      id: 9d3d0c7b-7131-4557-85b8-e32f167c1c38
+      version: -1
+      name: Evaluate remediation path rules
+      description: For a given alert and remediation path rules that are defined for that alert's attack surface rule, this takes each remediation path rule and looks at the rule criteria too see if the rule matches for the given alert. If multiple rules match, it will return the most recently created rule. This assumes that the rules passed in are filtered to correlate with the alert's attack su ...
```
```suggestion
      description: For a given alert and remediation path rules that are defined for that alert's attack surface rule, this takes each remediation path rule and looks at the rule criteria to see if the rule matches for the given alert. If multiple rules match, it will return the most recently created rule. This assumes that the rules passed in are filtered to correlate with the alert's attack surface rule.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Remediation_Path_Rules.yml:
```diff
@@ -0,0 +1,1010 @@
+      id: b9afa33d-f4d0-42b1-8676-bd3180067bbd
+      version: -1
+      name: What is the action of the matched rule?
+      description: Determines what is returned to the parent playbook based on contents of "ASM - Remediation Path Rule" field and other requirements.
```
```suggestion
      description: Determines what is returned to the parent playbook based on the contents of the "ASM - Remediation Path Rule" field and other requirements.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Remediation_Path_Rules_README.md:
```diff
@@ -0,0 +1,37 @@
+This playbook returns "RemediationAction" options based return from Remediation Path Rules API, or defaults to data collection task options from "Cortex ADM - Decision" subplaybook.
```
```suggestion
This playbook returns "RemediationAction" options based on the return from the Remediation Path Rules API, or defaults to data collection task options from the "Cortex ADM - Decision" sub-playbook.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Remediation_Path_Rules_README.md:
```diff
@@ -0,0 +1,37 @@
+This playbook uses the following sub-playbooks, integrations, and scripts.
+
+### Sub-playbooks
+* Cortex ASM - Decision
```
```suggestion
Cortex ASM - Decision
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Remediation_Path_Rules_README.md:
```diff
@@ -0,0 +1,37 @@
+* Cortex ASM - Decision
+
+### Integrations
+* CortexAttackSurfaceManagement
```
```suggestion
CortexAttackSurfaceManagement
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Remediation_Path_Rules_README.md:
```diff
@@ -0,0 +1,37 @@
+* RemediationPathRuleEvaluation
+
+### Commands
+* asm-list-remediation-rule
```
```suggestion
asm-list-remediation-rule
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Remediation_Path_Rules_README.md:
```diff
@@ -0,0 +1,37 @@
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| ExternallyDetectedProviders | Providers of external service. |  | Optional |
```
```suggestion
| ExternallyDetectedProviders | Providers of the external service. |  | Optional |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/README.md:
```diff
@@ -56,49 +66,79 @@ Playbook that looks at what ASM sub-type the alert is and directs it to differen
 ![Cortex ASM - SNMP Check](https://raw.githubusercontent.com/demisto/content/348579775c6c9b239d156f5398fd5c5a98e9458c/Packs/CortexAttackSurfaceManagement/doc_files/Cortex_ASM_-_SNMP_Check.png)
 
+#### Cortex ASM - Decision
+Playbook that returns "RemediationAction" options based on meeting "Automated Remediation Requirements" as well as whether ServiceNowV2 integration is setup.
```
```suggestion
Playbook that returns "RemediationAction" options based on meeting "Automated Remediation Requirements" as well as whether ServiceNowV2 integration is set up.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/README.md:
```diff
@@ -56,49 +66,79 @@ Playbook that looks at what ASM sub-type the alert is and directs it to differen
+![Cortex ASM - Decision](https://raw.githubusercontent.com/demisto/content/7b2f8db22f0be0cc55f8f0663894f459c3b3801e/Packs/CortexAttackSurfaceManagement/doc_files/Cortex_ASM_-_Decision.png)
+
+#### Cortex ASM - Remediation Path Rules
+Playbook that returns "RemediationAction" options based return from Remediation Path Rules API, or defaults to data collection task options from "Cortex ADM - Decision" subplaybook.
```
```suggestion
Playbook that returns "RemediationAction" options based on the return from the Remediation Path Rules API, or defaults to data collection task options from the "Cortex ADM - Decision" sub-playbook.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+
+#### Playbooks
+##### New: Cortex ASM - Decision
+- Created a new playbook that returns "Remediation Action" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is setup.
```
```suggestion
Created a playbook that returns "Remediation Action" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is set up.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Created a new playbook that returns "Remediation Action" options based on meeting "Automated Remediation Requirements" (https://github.com/demisto/content/tree/master/Packs/CortexAttackSurfaceManagement#automated-remediation-requirements) as well as whether ServiceNowV2 integration is setup.
+
+##### New: Cortex ASM - Remediation Path Rules
+- Created a new playbook that returns "Remediation Action" options based return from Remediation Path Rules API, or defaults to data collection task options from "Cortex ADM - Decision" subplaybook.
```
```suggestion
Created a playbook that returns "Remediation Action" options based on the return from the Remediation Path Rules API, or defaults to data collection task options from the "Cortex ADM - Decision" sub- playbook.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Created a new playbook that returns "Remediation Action" options based return from Remediation Path Rules API, or defaults to data collection task options from "Cortex ADM - Decision" subplaybook.
+
+##### Cortex ASM - ASM Alert
+- Updated the playbook to include the new "Cortex ASM - Remediation Path Rules" subplaybook to include the remediation path logic.
```
```suggestion
- Updated the playbook to include the new "Cortex ASM - Remediation Path Rules" sub-playbook to include the remediation path logic.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+
+##### Cortex ASM - ASM Alert
+- Updated the playbook to include the new "Cortex ASM - Remediation Path Rules" subplaybook to include the remediation path logic.
+- Updated the yml file to no longer include 'marketplaces' field.
```
```suggestion
- Remove the *marketplaces* field from the yml file.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Updated the yml file to no longer include 'marketplaces' field.
+
+##### Cortex ASM - CMDB Enrichment
+- Updated the yml file to no longer include 'marketplaces' field.
```
```suggestion
- Remove the *marketplaces* field from the yml file.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Updated the yml file to no longer include 'marketplaces' field.
+
+##### Cortex ASM - Detect Service
+- Updated the yml file to no longer include 'marketplaces' field.
```
```suggestion
Remove the *marketplaces* field from the yml file.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Updated the yml file to no longer include 'marketplaces' field.
+
+##### Cortex ASM - Enrichment
+- Updated the yml file to no longer include 'marketplaces' field.
```
Removed the *marketplaces* field from the yml file.

ShirleyDenkberg replied:
```suggestion
Removed the *marketplaces* field from the yml file.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Updated the yml file to no longer include 'marketplaces' field.
+
+##### Cortex ASM - GCP Enrichment
+- Updated the yml file to no longer include 'marketplaces' field.
```
```suggestion
Removed the *marketplaces* field from the yml file.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Updated the yml file to no longer include 'marketplaces' field.
+
+##### Cortex ASM - Remediation Guidance
+- Updated the yml file to no longer include 'marketplaces' field.
```
```suggestion
Removed the *marketplaces* field from the yml file.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Updated the yml file to no longer include 'marketplaces' field.
+
+##### Cortex ASM - Remediation
+- Updated the yml file to no longer include 'marketplaces' field.
```
```suggestion
Removed the *marketplaces* field from the yml file.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Updated the yml file to no longer include 'marketplaces' field.
+
+##### Cortex ASM - Tenable.io Enrichment
+- Updated the yml file to no longer include 'marketplaces' field.
```
```suggestion
Removed the *marketplaces* field from the yml file.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+
+#### Scripts
+##### GenerateASMReport
+- Updated the GenerateASMReport to include the new "ASM - Remediation Path Rule" field.
```
```suggestion
- Updated the GenerateASMReport to include the new *ASM - Remediation Path Rule* field.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Updated the GenerateASMReport to include the new "ASM - Remediation Path Rule" field.
+
+##### New: RemediationPathRuleEvaluation
+- Added the RemediationPathRuleEvaluation automation that checks for a matching remediation path rule (Xpanse only).
```
```suggestion
- Added the **RemediationPathRuleEvaluation** automation that checks for a matching remediation path rule (Xpanse only).
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+- Added the RemediationPathRuleEvaluation automation that checks for a matching remediation path rule (Xpanse only).
+
+#### Incident Fields
+- **ASM - Remediation Path Rule**
```
```suggestion
**ASM - Remediation Path Rule**
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+
+#### Integrations
+##### Cortex Attack Surface Management
+- Added the **asm-list-remediation-rule** command that returns list of remediation path rules (Xpanse only).
```
```suggestion
- Added the ***asm-list-remediation-rule*** command that returns a list of remediation path rules (Xpanse only).
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_5_1.md:
```diff
@@ -0,0 +1,48 @@
+#### Integrations
+##### Cortex Attack Surface Management
+- Added the **asm-list-remediation-rule** command that returns list of remediation path rules (Xpanse only).
+- Updated the yml file to no longer include 'marketplaces' field.
```
```suggestion
- Removed the *marketplaces* field from the yml file.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport.yml:
```diff
@@ -37,11 +37,14 @@ args:
 - description: Related or duplicate objects. (More of a nice to have because not sure how to track this).
   isArray: true
   name: asm_related
+- description: Matched remediation path rule (if there is a match).
```
Should this be:
A comma-separated list of remediation path rules to match.

johnnywilkes replied:
this is an array

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/README.md:
```diff
@@ -8,6 +8,11 @@ Generate an ASM Alert Summary report.
+## Used In
+---
+This script is used in the following playbooks and scripts.
+* Cortex ASM - ASM Alert
```
```suggestion
Cortex ASM - ASM Alert
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/README.md:
```diff
@@ -17,16 +22,17 @@ Generate an ASM Alert Summary report.
+| asm_data_collection | Collect information on data collection tasks. |
+| asm_tags | Tags from objects that can be used to determine other information \(if server is Dev for example\). |
+| asm_private_ip | Private IP addresses found. |
+| asm_related | Related or duplicate objects. \(More of a nice to have because not sure how to track this\). |
```
Should you really include the (more of a nice to have....) text?

johnnywilkes replied:
good catch, will remove

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/README.md:
```diff
@@ -17,16 +22,17 @@ Generate an ASM Alert Summary report.
+| asm_tags | Tags from objects that can be used to determine other information \(if server is Dev for example\). |
+| asm_private_ip | Private IP addresses found. |
+| asm_related | Related or duplicate objects. \(More of a nice to have because not sure how to track this\). |
+| asm_remediation_path_rule | Matched remediation path rule \(if there is a match\). |
```
Should this be:
A comma-separated list of remediation path rules to match.

johnnywilkes replied:
this is an array

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/README.md:
```diff
@@ -0,0 +1,26 @@
+For a given alert and remediation path rules that are defined for that alert's attack surface rule, this takes each remediation path rule and looks at the rule criteria too see if the rule matches for the given alert. If multiple rules match, it will return the most recently created rule. This assumes that the rules passed in are filtered to correlate with the alert's attack surface rule.
```
```suggestion
For a given alert and remediation path rules that are defined for that alert's attack surface rule, this script takes each remediation path rule and looks at the rule criteria to see if the rule matches for the given alert. If multiple rules match, it will return the most recently created rule. This assumes that the rules passed in are filtered to correlate with the alert's attack surface rule.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/README.md:
```diff
@@ -0,0 +1,26 @@
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| severity | Alert's Severity |
```
```suggestion
| severity | Alert's Severity. |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/README.md:
```diff
@@ -0,0 +1,26 @@
+| **Argument Name** | **Description** |
+| --- | --- |
+| severity | Alert's Severity |
+| ip | Alert's Remote IP |
```
```suggestion
| ip | Alert's Remote IP. |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/README.md:
```diff
@@ -0,0 +1,26 @@
+| severity | Alert's Severity |
+| ip | Alert's Remote IP |
+| development_environment | Is this in a development environment? |
+| cloud_managed | Is this Cloud Managed? |
```
```suggestion
| cloud_managed | Is this cloud managed? |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/README.md:
```diff
@@ -0,0 +1,26 @@
+| ip | Alert's Remote IP |
+| development_environment | Is this in a development environment? |
+| cloud_managed | Is this Cloud Managed? |
+| service_owner_identified | Has a Service Owner been Identified? |
```
```suggestion
| service_owner_identified | Has a service owner been identified? |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/README.md:
```diff
@@ -0,0 +1,26 @@
+| development_environment | Is this in a development environment? |
+| cloud_managed | Is this Cloud Managed? |
+| service_owner_identified | Has a Service Owner been Identified? |
+| tags | Includes Cloud and Xpanse tags |
```
```suggestion
| tags | Includes cloud and Xpanse tags |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/README.md:
```diff
@@ -0,0 +1,26 @@
+| cloud_managed | Is this Cloud Managed? |
+| service_owner_identified | Has a Service Owner been Identified? |
+| tags | Includes Cloud and Xpanse tags |
+| providers | Externally Detected Providers |
```
```suggestion
| providers | Externally detected providers. |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/README.md:
```diff
@@ -0,0 +1,26 @@
+| service_owner_identified | Has a Service Owner been Identified? |
+| tags | Includes Cloud and Xpanse tags |
+| providers | Externally Detected Providers |
+| remediation_path_rules | List of Remediation Path Rules for the Alert's Attack Surface Rule |
```
```suggestion
| remediation_path_rules | List of remediation path rules for the alert's attack surface rule. |
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.yml:
```diff
@@ -0,0 +1,39 @@
+args:
+- description: Alert's Severity
```
```suggestion
- description: Alert's severity.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.yml:
```diff
@@ -0,0 +1,39 @@
+args:
+- description: Alert's Severity
+  name: severity
+- description: Alert's Remote IP
```
```suggestion
- description: Alert's remote IP.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.yml:
```diff
@@ -0,0 +1,39 @@
+  name: ip
+- description: Is this in a development environment?
+  name: development_environment
+- description: Is this Cloud Managed?
```
```suggestion
- description: Is this cloud managed?
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.yml:
```diff
@@ -0,0 +1,39 @@
+- description: Is this Cloud Managed?
+  isArray: true
+  name: cloud_managed
+- description: Has a Service Owner been Identified?
```
```suggestion
- description: Has a service owner been identified?
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.yml:
```diff
@@ -0,0 +1,39 @@
+- description: Has a Service Owner been Identified?
+  isArray: true
+  name: service_owner_identified
+- description: Includes Cloud and Xpanse tags
```
Should this be:
A comma-separated list of cloud and Xpanse tags.

johnnywilkes replied:
this is an array

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/README.md:
```diff
@@ -0,0 +1,26 @@
+| development_environment | Is this in a development environment? |
+| cloud_managed | Is this Cloud Managed? |
+| service_owner_identified | Has a Service Owner been Identified? |
+| tags | Includes Cloud and Xpanse tags |
```
Should this be:
A comma-separated list of cloud and Xpanse tags.

johnnywilkes replied:
this is an array

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.yml:
```diff
@@ -0,0 +1,39 @@
+- description: Includes Cloud and Xpanse tags
+  isArray: true
+  name: tags
+- description: Externally Detected Providers
```
Should this be:
A comma-separated list of externally detected providers.

johnnywilkes replied:
this is an array

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/README.md:
```diff
@@ -0,0 +1,26 @@
+| cloud_managed | Is this Cloud Managed? |
+| service_owner_identified | Has a Service Owner been Identified? |
+| tags | Includes Cloud and Xpanse tags |
+| providers | Externally Detected Providers |
```
Should this be:
A comma-separated list of externally detected providers.

johnnywilkes replied:
this is an array

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.yml:
```diff
@@ -0,0 +1,39 @@
+- description: Externally Detected Providers
+  isArray: true
+  name: providers
+- description: List of Remediation Path Rules for the Alert's Attack Surface Rule
```
```suggestion
- description: List of remediation path rules for the alert's attack surface rule.
```

ShirleyDenkberg commented on Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.yml:
```diff
@@ -0,0 +1,39 @@
+- description: List of Remediation Path Rules for the Alert's Attack Surface Rule
+  isArray: true
+  name: remediation_path_rules
+comment: For a given alert and remediation path rules that are defined for that alert's attack surface rule, this takes each remediation path rule and looks at the rule criteria too see if the rule matches for the given alert. If multiple rules match, it will return the most recently created rule. This assumes that the rules passed in are filtered to correlate with the alert's attack surface rule ...
```
```suggestion
comment: For a given alert and remediation path rules that are defined for that alert's attack surface rule, this script takes each remediation path rule and looks at the rule criteria to see if the rule matches for the given alert. If multiple rules match, it will return the most recently created rule. This assumes that the rules passed in are filtered to correlate with the alert's attack surface rule.
```

ShirleyDenkberg commented:
@BigEasyJ @ilaner Doc review completed.

ilaner commented:
@johnnywilkes 
Please resolve conflicts from demisto/master
In addition, In the marketplace section let's specify which marketplace it supports. Leaving this field blank indicates it supports all marketplaces, including XSOAR.

johnnywilkes commented:
@ilaner , will fix them.  Also, FYI that this is a dependent on full merge of https://github.com/demisto/content/pull/24575

johnnywilkes commented:
@BigEasyJ , can you re-review please since I have made a few minor changes:

- https://github.com/demisto/content/pull/24380/commits/02383b84694a4489d1e297ece177855529e432a8 : removed marketplace tags
- https://github.com/demisto/content/pull/24380/commits/a4e8ce1e7ee1d2f1e48ea7e91d39c8d234f0e7db : change pack readme links
- https://github.com/demisto/content/pull/24380/commits/a8e55e453adcfa5ff610c4fbf92e1766b1d0bb80 : changed three fields to multiselect and updated stripChars

BigEasyJ reviewed (approved):
Reviewed updates and approved.

Here is also the [ticket in reference](https://jira-hq.paloaltonetworks.local/browse/CRTX-75275) to validation errors for changing the field types. Those errors should be ignored.

BigEasyJ commented:
Here is also the [ticket in reference](https://jira-hq.paloaltonetworks.local/browse/CRTX-75275) to validation errors for changing the field types. Those errors should be ignored.

johnnywilkes commented:
@BigEasyJ , thanks for the review

@ilaner , this has been approved by Xpanse. Team. There are build errors, but talking to @ShacharKidor and @dorschw , they said these could be bypassed with a "force merge."  The reason for the field type change was from XSOAR engineering in this ticket: https://jira-hq.paloaltonetworks.local/browse/CRTX-75275
We have done lots of testing and it doesn't seem like a breaking change.  Please let me know if you have any questions

ilaner reviewed (approved):
Good job
